pkg/instance/instance_integration_test.go

```diff
@@ -61,68 +71,143 @@ func TestInstanceHandler(t *testing.T) {
 	groupService := groupService{group: group}
 	stacks := stack.Stacks{
 		"whoami-go": stack.WhoamiGo,
+		"dhis2":     stack.DHIS2,
 	}
 	stackService := stack.NewService(stacks)
 	// classification 'test' does not actually exist, this is used to decrypt the stack parameters
 	helmfileService := instance.NewHelmfileService("../../stacks", stackService, "test")
 	instanceService := instance.NewService(instanceRepo, groupService, stackService, helmfileService)
 
+	s3Dir := t.TempDir()
+	s3Bucket := "database-bucket"
+	err = os.Mkdir(s3Dir+"/"+s3Bucket, 0o755)
+	require.NoError(t, err, "failed to create S3 output bucket")
+	s3 := inttest.SetupS3(t, s3Dir)
+	uploader := manager.NewUploader(s3.Client)
+	s3Client := storage.NewS3Client(s3.Client, uploader)
+	databaseRepository := database.NewRepository(db)
+	databaseService := database.NewService(s3Bucket, s3Client, groupService, databaseRepository)
+
+	authenticator := func(ctx *gin.Context) {
+		ctx.Set("user", &model.User{
+			ID:    user.ID, // TODO: Why is this necessary all of a sudden?
+			Email: "user1@dhis2.org",
+			Groups: []model.Group{
+				{
+					Name: "group-name",
+				},
+			},
+		})
+	}
 	client := inttest.SetupHTTPServer(t, func(engine *gin.Engine) {
 		var twoDayTTL uint = 172800
 		instanceHandler := instance.NewHandler(groupService, instanceService, twoDayTTL)
-		instance.Routes(engine, func(ctx *gin.Context) {
-			ctx.Set("user", user)
```

Comment:
Why did you replace that with the `authenticator` function from above?

Reply:
It's the same and I use it in two places now

Reply:
I want to understand why you had to change it? Because I think you don't have to deal with https://github.com/dhis2-sre/im-manager/pull/484#discussion_r1335490921 if you don't replace this function as the `user` we have already has an ID.